packages/hash/frontend/src/pages/[account-slug]/types/entity-type/property-type-form.tsx

```diff
@@ -1,60 +1,365 @@
+   */
+  const [titleValid, setTitleValid] = useState(false);
+  const [descriptionValid, setDescriptionValid] = useState(false);
+
+  return (
+    <Box
+      minWidth={500}
+      p={3}
+      component="form"
+      display="block"
+      onSubmit={handleSubmit}
     >
-      <TextField
-        label="Singular name"
-        required
-        placeholder="e.g. Stock Price"
-        defaultValue={initialTitle}
-      />
-      <TextField
-        multiline
-        inputProps={{ minRows: 1 }}
-        label={
-          <>
-            Description <QuestionIcon sx={{ order: 1, ml: 0.75 }} />
-          </>
-        }
-        required
-        placeholder="Describe this property type in one or two sentences"
-      />
-      <TextField
-        label="Expected values"
-        sx={{ alignSelf: "flex-start", width: "70%" }}
-        required
-        placeholder="Select acceptable values"
-      />
-    </Stack>
-    <Divider sx={{ mt: 2, mb: 3 }} />
-    <Stack direction="row" spacing={1.25}>
-      <Button {...createButtonProps} size="small">
-        Create new property type
-      </Button>
-      <Button {...discardButtonProps} size="small" variant="tertiary">
-        Discard draft
-      </Button>
-    </Stack>
-  </Box>
-);
+      <Stack
+        alignItems="stretch"
+        spacing={3}
+        sx={{
+          [`.${inputLabelClasses.root}`]: {
+            display: "flex",
+            alignItems: "center",
+          },
+        }}
+      >
+        <TextField
+          label="Singular name"
+          required
+          placeholder="e.g. Stock Price"
+          disabled={isSubmitting}
+          error={!!nameError}
+          helperText={nameError?.message}
+          success={titleValid}
+          {...register("name", {
+            required: true,
+            onChange() {
+              clearErrors("name");
+              setTitleValid(false);
+            },
+            async validate(value) {
+              const propertyTypeId = generateInitialPropertyTypeId(
+                generatePropertyTypeBaseUriForUser(value),
+              );
+
+              const res = await getPropertyType({ data: { propertyTypeId } });
+
+              const exists =
+                !res.data ||
+                !!getPersistedPropertyType(res.data, propertyTypeId);
+
+              if (getValues("name") === value && !exists) {
+                setTitleValid(true);
+              }
+
+              return exists ? "Property type name must be unique" : true;
+            },
+          })}
+        />
+        <TextField
+          multiline
+          inputProps={{ minRows: 1 }}
+          label={
+            <>
+              Description{" "}
+              <Tooltip
+                placement="top"
+                title="Descriptions help people understand what property types can be used for, and help make them more discoverable (allowing for reuse)."
+                PopperProps={{
+                  sx: {},
```

Comment:
is empty sx required here?

Reply:
[`174c282` (#1345)](https://github.com/hashintel/hash/pull/1345/commits/174c282e2a39f49c304a7b22711c1a40f2497dd1)